specificatie/nationaliteit.yaml

```diff
@@ -0,0 +1,88 @@
+openapi: 3.0.3
+info:
+  title: Nationaliteit definities
+  version: 2.0.0
+  contact: {}
+paths: {}
+components:
+  schemas:
+    AbstractNationaliteit:
+      type: object
+      required:
+        - type
+      properties:
+        type:
+          type: string
+        redenOpname:
+          $ref: "https://raw.githubusercontent.com/VNG-Realisatie/Haal-Centraal-common/v1.2.0/api-specificatie/common.yaml#/components/schemas/Waardetabel"
+      discriminator:
+        propertyName: type
+        mapping:
+          GbaNationaliteit: '#/components/schemas/GbaNationaliteit'
+          Nationaliteit: '#/components/schemas/Nationaliteit'
+          BehandeldAlsNederlander: '#/components/schemas/BehandeldAlsNederlander'
+          VastgesteldNietNederlander: '#/components/schemas/VastgesteldNietNederlander'
+    Nationaliteit:
+      description: |
+        * **redenOpname** : De reden op grond waarvan de persoon de nationaliteit gekregen heeft.
+      allOf:
+        - $ref: '#/components/schemas/AbstractNationaliteit'
+        - type: object
+          properties:
+            datumIngangGeldigheid:
+              $ref: 'datum.yaml#/components/schemas/AbstractDatum'
+            nationaliteit:
+              $ref: "https://raw.githubusercontent.com/VNG-Realisatie/Haal-Centraal-common/v1.2.0/api-specificatie/common.yaml#/components/schemas/Waardetabel"
+            inOnderzoek:
+              $ref: "#/components/schemas/NationaliteitInOnderzoek"
+    BehandeldAlsNederlander:
+      allOf:
+        - $ref: '#/components/schemas/AbstractNationaliteit'
+        - type: object
+          properties:
+            datumIngangGeldigheid:
```

Comment:
in dit geval wordt er dus geen veld behandeldAlsNederlander opgenomen (boolean) opgenomen? Dus de gebruiker moet afleiden dat er sprake is van behandeldAlsNederlander uit het type?

Reply:
een gebruiker (achter de proxy) krijgt dus een van de volgende opties:
```
        {
            "type": "Nationaliteit",
            "nationaliteit": {
                "code": "0027",
                "omschrijving": "Slowaakse"
            },
            "datumIngangGeldigheid": {
              "type": "Datum",
              "waarde": "2022-02-01"
            },
            "redenOpname": {
              "code": "301",
              "omschrijving": "Vaststelling bezit vreemde nationaliteit"
            }
        }
```

of

```
        {
            "type": "BehandeldAlsNederlander",
            "datumIngangGeldigheid": {
              "type": "Datum",
              "waarde": "2022-02-01"
            },
            "redenOpname": {
              "code": "310",
              "omschrijving": "Vaststelling bijzonder Nederlanderschap"
            }
        }
```

of

```
        {
            "type": "VastgesteldNietNederlander",
            "datumIngangGeldigheid": {
              "type": "Datum",
              "waarde": "2022-02-01"
            },
            "redenOpname": {
              "code": "310",
              "omschrijving": "Vaststelling bijzonder Nederlanderschap"
            }
        }
```

Reply:
klopt